pynest/examples/astrocytes/astrocyte_brunel_fixed_indegree.py

```diff
@@ -0,0 +1,389 @@
+---------------------------------------
+
+This script simulates a random balanced network with excitatory and inhibitory
+neurons and astrocytes. The astrocytes are modeled with ``astrocyte_lr_1994``,
+implemented according to [1]_, [2]_, and [3]_. The neurons are modeled with
+``aeif_cond_alpha_astro``, an adaptive exponential integrate-and-fire neuron
+supporting neuron-astrocyte interactions.
+
+The simulation results show how astrocytes affect neuronal excitability. The
+astrocytic dynamics, the slow inward current in the neurons induced by the
+astrocytes, and the raster plot of neuronal firings are shown in the created
+figures.
+
+In this version of the model, primary connections between populations are
+created with the fixed-indegree rule.
+
+References
+~~~~~~~~~~
+
+.. [1] Li, Y. X., & Rinzel, J. (1994). Equations for InsP3 receptor-mediated
+       [Ca2+]i oscillations derived from a detailed kinetic model: a
+       Hodgkin-Huxley like formalism. Journal of theoretical Biology, 166(4),
+       461-473. DOI: https://doi.org/10.1006/jtbi.1994.1041
+
+.. [2] De Young, G. W., & Keizer, J. (1992). A single-pool inositol
+       1,4,5-trisphosphate-receptor-based model for agonist-stimulated
+       oscillations in Ca2+ concentration. Proceedings of the National Academy
+       of Sciences, 89(20), 9895-9899. DOI:
+       https://doi.org/10.1073/pnas.89.20.9895
+
+.. [3] Nadkarni, S., & Jung, P. (2003). Spontaneous oscillations of dressed
+       neurons: a new mechanism for epilepsy?. Physical review letters, 91(26),
+       268101. DOI: https://doi.org/10.1103/PhysRevLett.91.268101
+
+See Also
+~~~~~~~~
+
+:doc:`astrocyte_small_network`
+
+"""
+
+###############################################################################
+# Import all necessary modules for simulation and plotting.
+
+import random
+
+import matplotlib.pyplot as plt
+import nest
+import numpy as np
+
+###############################################################################
+# Set simulation parameters.
+
+sim_params = {
+    "dt": 0.1,  # simulation resolution in ms
+    "pre_sim_time": 100.0,  # pre-simulation time in ms (data not recorded)
+    "sim_time": 1000.0,  # simulation time in ms
+    "N_rec_spk": 100,  # number of neurons to record from with spike recorder
+    "N_rec_mm": 50,  # number of nodes (neurons, astrocytes) to record from with multimeter
+    "n_threads": 4,  # number of threads for NEST
+    "seed": 100,  # seed for the random module
+}
+
+###############################################################################
+# Set network parameters.
+
+network_params = {
+    "N_ex": 8000,  # number of excitatory neurons
+    "N_in": 2000,  # number of inhibitory neurons
+    "N_astro": 10000,  # number of astrocytes
+    "CE": 800,
+    "CI": 200,
+    "p_third_if_primary": 0.5,  # probability of each created neuron-neuron connection to be paired with one astrocyte
+    "pool_size": 10,  # astrocyte pool size for each target neuron
```

Comment:
Good question. This is something that might be confusing to the users and we should document it somewhere.

The pool_size parameter indeed limits the number of astrocytes that can be connected to a particular postsynaptic neuron. However, the algorithm could create several connections from the selected astrocyte to its postsynaptic neuron partner. This happens because we have to map tripartite neuron-astrocyte interactions into binary connectivity scheme, which is not ideal. 

Suppose we have two presynaptic neurons N1 and N2, one astrocyte A, and one post-synaptic neuron P. Suppose we have synapses S1 (N1 -> P) and S2 (N2 ->P), and suppose A interacts with both of those synapses. The NEST connectivity function will create the following:

synapses:
S1: N1 -> P 
S2: N2 -> P  

neuron-astrocyte interactions:
N1 -> A (for S1)
N2 -> A (for S2)

astrocyte-neuron interactions:
A -> P (for S1)
A -> P (for S2)

In principle, these two astrocyte-neuron interactions (for S1 and for S2) might have different parameters and different dynamics. However, if you look only at the presence/absence of a connection, you would just see double connection between A and P.

Reply:
Yes, multiple connections between an astrocyte and a neuron is allowed. Currently the TripartiteConnect() is documented [here](https://nest-simulator.readthedocs.io/en/latest/synapses/connectivity_concepts.html#tripartite-connectivity). It can be seen from the figure that the `pool_size` limits the number of astrocytes for each target neuron, but not the number of connections. But maybe it is right that the documentation can be improved. Maybe in another PR though.

However, this reminds me of another issue: the documentations relevant to TripartiteConnect(), including the astrocyte example scripts, will probably be out-dated with this PR, because the function interface will change. @heplesser should we deal with this in this PR? If yes, maybe I can try to update the documentation and pull a PR to your PR, or anything.

Reply:
> Yes, multiple connections between an astrocyte and a neuron is allowed. Currently the TripartiteConnect() is documented [here](https://nest-simulator.readthedocs.io/en/latest/synapses/connectivity_concepts.html#tripartite-connectivity). It can be seen from the figure that the `pool_size` limits the number of astrocytes for each target neuron, but not the number of connections. But maybe it is right that the documentation can be improved. Maybe in another PR though.
> 
> However, this reminds me of another issue: the documentations relevant to TripartiteConnect(), including the astrocyte example scripts, will probably be out-dated with this PR, because the function interface will change. @heplesser should we deal with this in this PR? If yes, maybe I can try to update the documentation and pull a PR to your PR, or anything.

Correction: I just realized this has been taken care of, except that maybe [this one](https://nest-simulator.readthedocs.io/en/latest/synapses/connectivity_concepts.html#tripartite-bernoulli-with-pool) still needs to be changed (in doc/htmldoc/synapses/connectivity_concepts.rst).

Reply:
I have now updated the tripartite part of the Connectivity Concepts documentation. The new version should become [available on Readthedocs as a preview](https://nest-simulator--3204.org.readthedocs.build/en/3204/synapses/connectivity_concepts.html) within an hour. I decided to not repeat three almost identical code examples, since that makes it difficult for readers to see how the examples differ. I only kept the lines with differences in examples B and C.